File: wlr/SPX.Util/IntStorage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SPX.Util
{
    public class IntStorage
    {
        private static readonly Dictionary<float, int> __slots = new Dictionary<float, int>();

        public static int Get(float value)
        {
            if (!__slots.ContainsKey(value))
            {
                __slots.Add(value, (int)value);
            }
            return __slots[value];
        }
    }
}

File: port-workspace/spx-ported/src/com/spx/util/IntStorage.java
package com.spx.util;import java.util.*;
    public class IntStorage
    {
        private static HashMap<Float, Integer> __slots = new HashMap<Float, Integer>();

        public static int Get(float value)
        {
            if (!__slots.containsKey(value))
            {
                __slots.put(value, (int)value);
            }
            return __slots.get(value);
        }
    }
